action/protocol/execution/protocol.go

```diff
@@ -75,15 +75,12 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
 	return receipt, nil
 }
 
-// Validate validates an execution
+// Validate validates the size of an execution
 func (p *Protocol) Validate(_ context.Context, act action.Action, _ protocol.StateReader) error {
-	exec, ok := act.(*action.Execution)
-	if !ok {
-		return nil
-	}
-	// Reject oversize execution
-	if exec.TotalSize() > ExecutionSizeLimit {
-		return errors.Wrap(action.ErrActPool, "oversized data")
+	if exec, ok := act.(*action.Execution); ok {
+		if exec.TotalSize() > ExecutionSizeLimit {
+			return errors.Wrap(action.ErrActPool, "oversized data")
+		}
```

Comment:
let's use the current code `if !ok { return nil }`, this is an early exit style used very commonly in our code

think about if future we need to add another validation check, like gas amount, the early exit style is better than changing to a big if {} statement

Reply:
good suggestions